aios/trajectory.py

```diff
@@ -0,0 +1,186 @@
+    and shooter launch speed to send the note into the speaker.
+
+    The origin has been chosen as the center of the speaker-back that is being scored into. "x-axis" is parallel to side that speaker is attached to.
+
+    ...
+
+    Attributes
+    ----------
+    x : float
+        The x-coordinate of the robot at the time of launch
+    y: float
+        The y-coordinate of the robot at the time of launch.
+    vx: float
+        The change in the x-coordinate with respect to time at launch time.
+    vy : float
+        The change in the y-coordinate with respect to time at launch time.
+    phi: float
+        The current angle that the shooter is at with respect to the y-axis
+    v : float
+        The current launch speed of the shooter.
+    theta : float
+        The angle relative to the horizontal that the shooter is inclined at. 
+    angleWeight : float
+        The weight in the loss function of a change in angle
+    speedWeight : float
+        The weight in the loss function of a change in shooter launching speed.
+    headingWeight : float
+        The weight in the loss function of a change in shooter heading.
+
+
+    Methods 
+    -------
+    getNewShooterState()
+        Returns a dict of a new state the motor should be in. (Launch speed and angle of inclination relative to floor).
+    """
+    def __init__(self, x:float, y:float, vx:float, vy:float, phi:float, v : float, theta:float, angleWeight:float, speedWeight:float, headingWeight:float):
+        """
+        Parameters
+        ----------
+        x : float
+            The x-coordinate of the robot at the time of launch.
+        y: float
+            The y-coordinate of the robot at the time of launch.
+        vx: float
+            The change in the x-coordinate with respect to time at launch time.
+        vy : float
+            The change in the y-coordinate with respect to time at launch time.
+        phi: float
+            The current angle that the shooter is at with respect to the y-axis. Towards the left side is negative and towards the right is positive (viewing the opening of the speaker)
+        launchSpeed : float
+            The current launch speed of the shooter.
+        currentAngleState : float
+            The angle relative to the horizontal that the shooter is inclined at. 
+        angleWeight : float
+            The weight in the loss function of a change in angle.
+        speedWeight : float
+            The weight in the loss function of a change in shooter launching speed.
+        headingWeight : float
+            The weight in the loss function of a change in shooter heading.
+
+        """
+
+        self.x = x
+        self.y = y
+        self.vx = vx
+        self.vy = vy
+        self.phi = phi
+        self.v = v
+        self.theta = theta
+        self.angleWeight = angleWeight
+        self.speedWeight = speedWeight
+        self.headingWeight = headingWeight
+
+        self.MIN_HEIGHT = 1.98 #m
```

Comment:
since these constants are not actually specific to any one trajectory, could you make these just like general constants? (like static variables in java)
I believe in python this would be just the same thing you have, but you don't use self.

Reply:
Side note, I'm feeling a lot of hatred towards python right now